### ISSUE TITLE
Moved images break some functions 

### ISSUE THREAD

**Issue by tobias47n9e** [nicolas-raoul/apps-android-commons#224]
I think that moved images break the link to the app. For example the download button will show a download failed notification. Description and categories are those from the initial upload.

Version 1.24

**Comment by nicolas-raoul** [comment]
What do you mean by "moved images"?

**Comment by tobias47n9e** [comment]
Best way to reproduce is to upload an image with the app, Then move it on commons, and then view it in the app after the move is complete.

**Comment by nicolas-raoul** [comment]
I thought only admin could move an image on Commons. It is not a usual operation, right? I guess it breaks all pages that use the image.

Could you please post a screenshot of each part of the app which breaks, along with how it should behave instead? Thanks!

**Comment by tobias47n9e** [comment]
Yes, moving images takes special rights. The response is probably a redirect, which the app could use to update the link?

The image detail screen:
<img src="https://cloud.githubusercontent.com/assets/6953323/17563232/b8332cd6-5f2e-11e6-864c-ee8fc31ce2a7.png" width="200">

No categories ("Download" does not work):
<img src="https://cloud.githubusercontent.com/assets/6953323/17563245/c449573e-5f2e-11e6-85b1-8db2243e75f0.png" width="200">

"Open in browser" shows redirect:
<img src="https://cloud.githubusercontent.com/assets/6953323/17563248/c92bd542-5f2e-11e6-996e-d07f9aec2758.png" width="200">

The image on Commons (note the existing categories):
https://commons.wikimedia.org/wiki/File:Cichorium_intybus_blossom.jpg

**Comment by neslihanturan** [comment]
Hard to test this issue since I am not authorized to move images. But I suspect this is only because of we fetch uploads from database, not from server each time. There is a relevant issue that I will reference.
